database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/PostCell.swift

```diff
@@ -32,7 +32,17 @@ struct PostCell: View {
             #if os(iOS) || os(macOS)
               starButton
                 .onTapGesture {
-                  post.didTapStarButton()
+                  if #available(iOS 15, macOS 12, *) {
+                    #if swift(>=5.5)
+                      Task { await
```

Comment:
Why is a `Task` needed here? Can it be just `await post.didTapStarButtonAsync()`?

Reply:
Since the code is executed within`.onTapGesture`, which does not have an async signature, we need to use Task to open the async gesture. So if I remove the `Task` it will result in an error message: `Cannot pass function of type '() async -> Void' to parameter expecting synchronous function type`